### DISCUSSION TITLE
shallow clone of git dependencies

### DISCUSSION THREAD

**Discussion by trusktr** [npm/feedback#99] [CLI] [answered]
Some git repos have a ridiculously large history and take forever to clone.

Does `npm` shallow clone dependencies? If not, it would be nice to have an option for that.

**Comment by MylesBorins** [comment]
hey @trusktr thanks for the suggestion. pinging @npm/cli to chime in on this one.

**Comment by MylesBorins** [comment, marked as answer]
npm does indeed shallow clone dependencies from git repository hosts that we know support it.

e.g. GitHub + gitlab + bitbucket

We don't do this for arbitrary / unknown git host a shallow cloning is not a default feature of a git host.

One thing we could explore would be trying to shallow clone first and falling back to full clone if it fails.
